### PR TITLE
add .dockstore.yml publish documentation

### PR DESCRIPTION
**Pull request by GFJHogue**
Main ticket: https://github.com/dockstore/dockstore/issues/4100
Relevant feature PR: https://github.com/dockstore/dockstore/pull/4085

Adding documentation for setting workflow/service publish state via `.dockstore.yml`.

### CODE REVIEW

**Review by coverbeck — APPROVED** [review]
Looks good, but should we merge it into develop, which is our public view? People might start using it and expecting it to work -- apparently NF-Core already has.

Options:

1. Create a 1.11 branch and merge it there, then merge 1.11 when we release 1.11
2. Label this as a 1.11 feature inline in the doc (looks like that it will be tricky in this case).

**Comment by GFJHogue** [comment]
> Looks good, but should we merge it into develop, which is our public view?
> ...
> 1. Create a 1.11 branch and merge it there, then merge 1.11 when we release 1.11

I agree, but this feels like we'd be treating `develop` like how we treat `master` everywhere else.

Why is `develop` the public view in the first place for dockstore-documentation?

**Comment by coverbeck** [comment]
> Why is `develop` the public view in the first place for dockstore-documentation?

I was wondering that myself as I typed in my initial comment. :) I think it was set up that way in the beginning and we never revisited it.

Seems like should follow the more standard Hubflow model. But now we have links to develop from the outside, so we might be stuck with it, unless readthedocs lets you do redirects.

**Comment by GFJHogue** [comment]
> I was wondering that myself as I typed in my initial comment. :) I think it was set up that way in the beginning and we never revisited it.
> Seems like should follow the more standard Hubflow model. But now we have links to develop from the outside, so we might be stuck with it, unless readthedocs lets you do redirects.

That makes sense. I'll go ahead and make a 1.11 branch for this PR, which we can merge in on release.

**Comment by denis-yuen** [comment]
> Seems like should follow the more standard Hubflow model. But now we have links to develop from the outside, so we might be stuck with it, unless readthedocs lets you do redirects.

Did you have particular links to develop in mind? I wonder if they are under our control/influence

**Comment by coverbeck** [comment]
> Did you have particular links to develop in mind? I wonder if they are under our control/influence

Hmm, I may have assumed wrong. I was thinking of some of @bethsheets documents, although she brought one into our docs, so that might be moot. So it might be safe to change after all.

File: docs/assets/templates/.dockstore.yml [hunk 1]
```diff
@@ -9,6 +9,11 @@ service:
   author:
   description:
 
+  # `publish` is a service-wide setting and will affect ALL branches/tags; only set this as needed in a main branch.
```
Comment (denis-yuen) [inline docs/assets/templates/.dockstore.yml:12]: maybe `dockstore-wide`
service is like tool or workflow for us

**Review by denis-yuen — APPROVED** [review]
minor comment change

File: docs/assets/templates/.dockstore.yml (hunk 1 above)
Comment (GFJHogue) [reply docs/assets/templates/.dockstore.yml:12]: I indeed mean service-wide here, just as I say workflow-wide elsewhere in this PR.

Maybe there's a better way to word it?

Comment (denis-yuen) [reply docs/assets/templates/.dockstore.yml:12]: Oh no, I just missed that this is legit the one for services, carry on

**Review by denis-yuen — APPROVED** [review]

**Review by Richard-Hansen — APPROVED** [review]

**Review by NatalieEO — APPROVED** [review]

**Review by garyluu — APPROVED** [review]

**Pull request merged**